agent/engine/docker_task_engine.go

```diff
@@ -731,17 +731,20 @@ func (engine *DockerTaskEngine) concurrentPull(task *apitask.Task, container *ap
 
 	// Record the task pull_started_at timestamp
 	pullStart := engine.time().Now()
-	defer func(startTime time.Time) {
-		seelog.Infof("Task engine [%s]: Finished pulling container %s in %s",
-			task.Arn, container.Image, time.Since(startTime).String())
-	}(pullStart)
 	ok := task.SetPullStartedAt(pullStart)
 	if ok {
 		seelog.Infof("Task engine [%s]: Recording timestamp for starting image pulltime: %s",
 			task.Arn, pullStart)
 	}
-
-	return engine.pullAndUpdateContainerReference(task, container)
+	metadata := engine.pullAndUpdateContainerReference(task, container)
+	if (metadata.Error == nil) {
+		seelog.Infof("Task engine [%s]: Finished pulling container %s in %s",
+			task.Arn, container.Image, time.Since(pullStart).String())
+	} else {
```

Comment:
is `metadata.Error` a string?

Reply:
No it is not. It is of type apierrors.NamedError where 
type NamedError interface {
	error
	ErrorName() string
}

Reply:
No it is not a string. It is of type apierrors.NamedError where
type NamedError interface {
error
ErrorName() string
}